src/main/java/com/google/sps/servlets/CreateQuestionServlet.java

```diff
@@ -0,0 +1,149 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+import com.google.sps.data.QuestionClass;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.*;
+import java.util.Date;
+import javax.servlet.ServletException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+/** Servlet that creates and stores questions
+* @author Klaudia Obieglo 
+*/
+@WebServlet("/createQuestion")
+public class CreateQuestionServlet extends HttpServlet{
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /* Servlet Receives information from the client about the question they want to save */
+    Long date = (new Date()).getTime();
+    String question = getParameter(request, "question", "");
+    String marks = getParameter(request, "marks", "");
+    
+    UserService userService = UserServiceFactory.getUserService();
+    String ownerID = userService.getCurrentUser().getEmail(); 
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Create a Question Entity with the parameters provided
+    Entity questionEntity = new Entity("Question");
+    questionEntity.setProperty("question",question);
+    questionEntity.setProperty("marks",marks);
+    questionEntity.setProperty("date",date);
+    questionEntity.setProperty("ownerID",ownerID);
+    try{
+      datastore.put(questionEntity);    
+    }catch (DatastoreFailureException e){
+      System.out.println("Datastore is not responding right now. Try Again Later");
+    }
+    addQuestionToExamList(questionEntity.getKey().getId(),ownerID);
```

Comment:
addQuestionToExamList() will be executed even if DatastoreFailureException is thrown. Is that by design? What would getKey() return in this case?

Reply:
It is by bad design. It should only be executed when DataStore works. In this case getKey() would return null as the entity is not placed in the datastore. I will fix that now